src/v2/controllers/vreplicaset_controller/proof/guarantee.rs

```diff
@@ -0,0 +1,258 @@
+use crate::kubernetes_api_objects::spec::prelude::*;
+use crate::kubernetes_cluster::spec::{cluster::*, message::*};
+use crate::reconciler::spec::io::*;
+use crate::temporal_logic::{defs::*, rules::*};
+use crate::vreplicaset_controller::{
+    model::{install::*, reconciler::*},
+    proof::helper_invariants,
+    trusted::{
+        rely_guarantee::*,
+        spec_types::*, 
+    },
+};
+use crate::vstd_ext::seq_lib::*;
+use vstd::prelude::*;
+
+verus! {
+
+spec fn stronger_vrs_guarantee_delete_req(req: DeleteRequest) -> StatePred<ClusterState> {
```

Comment:
Could you add a comment to explain this invariant?

Reply:
I improved the comment.